### PR TITLE
Chest Value: Only show minion's generated stuff and visual upgrade

### PR DESCRIPTION
**Pull request by viciscat**
[video here wink wink nudge nudge punch punch shoot shoot](https://discord.com/channels/879732108745125969/1249103983499411506/1264994412698800271)
video kinda outdated, tweaks happened

### CODE REVIEW

File: src/main/java/de/hysky/skyblocker/skyblock/ChestValue.java [hunk 1]
```diff
@@ -42,35 +51,54 @@ public static void init() {
...
+							.tooltip(minion ? Tooltip.of(Text.translatable("skyblocker.config.general.minionValue.@Tooltip")) : Tooltip.of(Text.translatable("skyblocker.config.general.chestValue.@Tooltip")))
 							.build()
 					);
 				}
 			}
 		});
 	}
 
-	private static Text getDungeonChestProfit(GenericContainerScreenHandler handler, Text title, String titleString) {
+	private static void addValueToContainer(GenericContainerScreen genericContainerScreen, Text chestValue, Text title) {
```
Comment (kevinthegreat1) [inline src/main/java/de/hysky/skyblocker/skyblock/ChestValue.java:76]: Can you move this metho all the way to the end (before `ScrollingTextWidget`) cause this method is called last.

File: src/main/java/de/hysky/skyblocker/skyblock/ChestValue.java [hunk 2]
```diff
@@ -138,19 +166,42 @@ private static Text getDungeonChestProfit(GenericContainerScreenHandler handler,
...
+						(coinsLine = ItemUtils.getLoreLineIf(stack, s -> s.contains("Held Coins:"))) != null) {
+
+					String source = coinsLine.split(":")[1];
+					try {
+						value += DecimalFormat.getNumberInstance(java.util.Locale.US).parse(source.trim()).doubleValue();
+					} catch (ParseException e) {
+						LOGGER.warn("[Skyblocker] Failed to parse {}", source);
+					}
+
+				}
```
Comment (kevinthegreat1) [inline src/main/java/de/hysky/skyblocker/skyblock/ChestValue.java:201]: I would separate all of this into a separate method, like `getMinionSlots`.

**Review by kevinthegreat1 — CHANGES_REQUESTED** [review]
Thanks for finally rendering it properly. Making it no longer update is a bit of an inconvenience. There's two ways you could make it update. 1: Find the scrolling text widget in `addValueToContainer` and update it. You should also give scrolling text widget class a different name since it's specific to chest value now. 2: Another field in `HandledScreenMixin`, the bloated savior.

File: src/main/java/de/hysky/skyblocker/skyblock/ChestValue.java [hunk 3]
```diff
@@ -42,35 +51,54 @@ public static void init() {
 				String titleString = title.getString();
 				if (DUNGEON_CHESTS.contains(titleString)) {
 					if (SkyblockerConfigManager.get().dungeons.dungeonChestProfit.enableProfitCalculator) {
-						ScreenEvents.afterTick(screen).register(screen_ ->
-								((ScreenAccessor) screen).setTitle(getDungeonChestProfit(genericContainerScreen.getScreenHandler(), title, titleString))
-						);
+                        Text dungeonChestProfit = getDungeonChestProfit(genericContainerScreen.getScreenHandler());
+                        if (dungeonChestProfit != null) addValueToContainer(genericContainerScreen, dungeonChestProfit, title);
```
[the remark this answers is not in the document]
Comment (viciscat) [reply src/main/java/de/hysky/skyblocker/skyblock/ChestValue.java:55]: yea but it re-added the widgets, i could make it remove all widgets that are an instance of ScrollingTextWidget but eh. Or I could add an ScreenHandlerListenerOrWhateverItsNameWas and update with setMessage() when a slot changes.

File: src/main/java/de/hysky/skyblocker/skyblock/ChestValue.java [hunk 4]
```diff
@@ -42,35 +51,54 @@ public static void init() {
...
+					boolean minion = MINION_PATTERN.matcher(title.getString().trim()).find();
 					Screens.getButtons(screen).add(ButtonWidget
 							.builder(Text.literal("$"), buttonWidget -> {
 								Screens.getButtons(screen).remove(buttonWidget);
-								ScreenEvents.afterTick(screen).register(screen_ ->
-										((ScreenAccessor) screen).setTitle(getChestValue(genericContainerScreen.getScreenHandler(), title, titleString))
-								);
+									Text chestValue = getChestValue(genericContainerScreen.getScreenHandler(), minion);
+                                    if (chestValue != null) {
+                                        addValueToContainer(genericContainerScreen, chestValue, title);
```
[the remark this answers is not in the document]
Comment (viciscat) [reply src/main/java/de/hysky/skyblocker/skyblock/ChestValue.java:64]: same as above

File: src/main/java/de/hysky/skyblocker/skyblock/ChestValue.java (hunk 2 above)
Comment (viciscat) [reply src/main/java/de/hysky/skyblocker/skyblock/ChestValue.java:201]: that returns the slots or the coins 🤔

File: src/main/java/de/hysky/skyblocker/skyblock/ChestValue.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (viciscat) [reply src/main/java/de/hysky/skyblocker/skyblock/ChestValue.java]: i did the former

File: src/main/java/de/hysky/skyblocker/skyblock/ChestValue.java (hunk 2 above)
Comment (kevinthegreat1) [reply src/main/java/de/hysky/skyblocker/skyblock/ChestValue.java:189]: The slots seem easier and nicer.

File: src/main/java/de/hysky/skyblocker/skyblock/ChestValue.java [hunk 5]
```diff
@@ -193,4 +232,58 @@ static Text getValueText(long value, boolean hasIncompleteData) {
...
+		int x = ((HandledScreenAccessor) genericContainerScreen).getX();
+		((ScreenAccessor) genericContainerScreen).setTitle(Text.empty());
+		TextRenderer textRenderer = MinecraftClient.getInstance().textRenderer;
+		int chestValueWidth = Math.min(textRenderer.getWidth(chestValue), Math.max((backgroundWidth - 8) / 2 - 2, backgroundWidth - 8 - textRenderer.getWidth(title)));
+
+		TextWidget chestValueWidget = new ChestValueTextWidget(chestValueWidth, textRenderer.fontHeight, chestValue, textRenderer);
+		chestValueWidget.setPosition(x + backgroundWidth - chestValueWidget.getWidth() - 4, y + 6);
+		Screens.getButtons(genericContainerScreen).add(chestValueWidget);
+
+		ChestValueTextWidget chestTitleWidget = new ChestValueTextWidget(backgroundWidth - 8 - chestValueWidth - 2, textRenderer.fontHeight, title.copy().fillStyle(Style.EMPTY.withColor(4210752)), textRenderer);
```
Comment (kevinthegreat1) [inline src/main/java/de/hysky/skyblocker/skyblock/ChestValue.java:249]: Wait I’m confused. Why are you rendering the title? Vanilla does that already?

**Review by kevinthegreat1 — CHANGES_REQUESTED** [review]

File: src/main/java/de/hysky/skyblocker/skyblock/ChestValue.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (kevinthegreat1) [reply src/main/java/de/hysky/skyblocker/skyblock/ChestValue.java]: Oh I didn’t notice you had two instances of text widget per screen. You could do the second option much easier if there was only one.

File: src/main/java/de/hysky/skyblocker/skyblock/ChestValue.java (hunk 5 above)
Comment (viciscat) [reply src/main/java/de/hysky/skyblocker/skyblock/ChestValue.java:249]: to shrink the title if need be, cuz if the title already takes 95% of the title space then you won't be seeing the coins x)
to see what i mean test in an enchanted sack or check the video

**Review by kevinthegreat1 — APPROVED** [review]

**Pull request merged**